.release-notes/4127.md

```diff
@@ -0,0 +1,13 @@
+## Fix for infinite Ranges
+
+Even though all Range objects that contain NaN parameters are considered infinite, some did not iterate at all or produced an error after the first iteration. For example, the code from the Range documentation: 
```

Comment:
I think this fixes more ranges than only NaN / floating point ranges, so I would suggest including that, too. For example, this one will never iterate:

```pony
var idx: USize = 0
for i in Range(0, 1, 0) do
  idx = idx + 1
end
env.out.print(idx.string()) // prints "0"
```

Reply:
Yes, this was also stated in the discussion above but not properly in the release note portion.